protostar/commands/deploy/deploy_command.py

```diff
@@ -19,13 +19,24 @@ class CompilationOutputNotFoundException(ProtostarException):
 class DeployCommand(Command):
     gateway_url_arg = Command.Argument(
         name="gateway_url",
-        description="The URL of a StarkNet gateway.",
+        description="The URL of a StarkNet gateway. It is required unless `--network` is provided.",
         type="str",
     )
 
     network_arg = Command.Argument(
         name="network",
-        description="The name of the StarkNet network.",
+        short_name="n",
+        description=(
+            "\n".join(
+                [
+                    "The name of the StarkNet network.",
+                    "It is required unless `--gateway_url` is provided.",
```

Comment:
It's a mistake, we probably should have`gateway-url` argument. It was probably reason of your problems with deploying on devnet :p